test-utils/state-viewer/src/main.rs

```diff
@@ -382,9 +332,41 @@ fn main() {
                 }
             }
         }
-        ("dump_state", Some(args)) => {
-            let output_json = args.is_present("json");
-            dump_state(&PathBuf::from(home_dir), store, near_config, output_json);
+        ("dump_state", Some(_args)) => {
+            let (runtime, state_roots, height) = load_trie(store.clone(), home_dir, &near_config);
+            let home_dir = PathBuf::from(&home_dir);
+            let mut chain_store = ChainStore::new(store.clone());
+            let genesis_hash = chain_store.get_block_hash_by_height(0).unwrap();
+            println!("Genesis hash: {}", &genesis_hash);
```

Comment:
Here we use the hash of genesis the chain had before the dump - means we can't really compare it to contents of the dump.
Seems like the possible options are:
- create a genesis block here in the code and compute its hash (but could be non-trivial @bowenwang1996 )
- just hash the files we dumped. Could be done inside `new-genesis-from-existing-state.sh` script

Reply:
Good point on the second point. We can go with that (I suggested the first but yeah it's more work).

Reply:
second approach do you mean
- dump
- start node from new genesis.json
- we know the new genesis block hash now

Reply:
Second is to use file hash instead of genesis block hash. It doesn't let us compare different ways of dumping state to each other, but can still guarantee we download the expected data.

Reply:
File hash is not as ideal as genesis block hash, but since you suggest do it in `new-genesis-from-existing-state.sh` script I can easilly do:
1. state-viewer dump_state, modify this to dump only, doesn't try to calculate new genesis block hash
2. we now have a new output_records.json and output_config.json.
3. script boot a new node from new output_records/config and it will calculate the new genesis block hash
4. script can now rename output_records with new genesis block hash.

Reply:
@mikhailOK I think you're right. I'd have to copy paste a lot of code to fit "print genesis hash" into an existing subcommand or in a new subcommand, and finally it looks more ugly than your first option. And it's worse than it because it goes to main near binary. Will see if it's easy to clean up it to state-viewer (which is exactly as first option)

Reply:
@bowenwang1996 sorry somehow i forgot to push yesterday's change (regarding `check when the node computes genesis state that the hashes match`) pushed now

Reply:
@mikhailOK Fixed follow first option. Hash after this change and the genesis_hash I commit happens to be same because I run a node from testnet genesis.json and not reach min_num_peers, it didn't produce new block